cli/src/mcp/tools/searchFiles.ts

```diff
@@ -0,0 +1,142 @@
+import fs from "fs";
+import { glob } from "glob";
+import path from "path";
+import { z } from "zod";
+
+import { normalizeError } from "../../utils/errors.js";
+import type { McpTool } from "../types/tools.js";
+
+export class SearchFilesTool implements McpTool {
+  name = "search_files";
+  description = "Search for files matching a pattern";
+
+  inputSchema = z.object({
+    pattern: z
+      .string()
+      .describe(
+        "File matching pattern using glob syntax. Examples: '**/*.py' (all Python files), 'docs/*.md' (Markdown in docs), '*.{js,ts}' (JS/TS files)"
+      ),
+
+    directory: z
+      .string()
+      .optional()
+      .describe(
+        "Optional: Target directory for the search operation. Directory must be an absolute directory. When not provided, searches from the current working directory"
+      ),
+
+    case_sensitive: z
+      .boolean()
+      .optional()
+      .describe(
+        "Optional: Controls whether pattern matching respects character case. Set to true for exact case matching, false otherwise (default: false)"
+      ),
+
+    limit: z
+      .number()
+      .int()
+      .positive()
+      .optional()
+      .describe(
+        "Optional: Controls the maximum count of files included in search results. Prevents overwhelming output for large directories (default: 100)"
+      ),
+
+    sort_by_modified: z
+      .boolean()
+      .optional()
+      .describe(
+        "Optional: Orders search results by file modification timestamp. When enabled, most recently modified files appear first (default: false)"
+      ),
+  });
+
+  async execute({
+    pattern,
+    directory = process.cwd(),
+    case_sensitive = false,
+    limit = 100,
+    sort_by_modified = false,
+  }: z.infer<typeof this.inputSchema>) {
+    try {
+      // Use glob for proper glob pattern support
+      const globOptions = {
```

Comment:
this is a good default, but it would be pretty cool to respect .gitignore when available